### DISCUSSION TITLE
What is a reasonable value for DBH for grass PFTs?

### DISCUSSION THREAD

**Discussion by Aariq** [EDmodel/ED2#383] [Q&A] [answered]
In designing a .css file to use with `IED_INIT_MODE=6`, I'm not sure what a reasonable value for DBH for a grass PFT is.  I had tried 0.6, but that resulted in many convergence errors.  Increasing it to 1 seemed to fix that, but I still have no idea if that's a reasonable number.  I assume DBH isn't interpreted literally (since grasses are typically not at breast height), but that it is used in some allometry equations.  How might I figure out what a reasonable value is?

**Comment by mpaiao** [comment, marked as answer]
You are right about the DBH not having a literal meaning for grasses, but the value is still used for computing allometric relationships. For any PFT, the minimum DBH depends on the minimum height allowed for the PFT (parameter hit_min). If you are using the default ED2 parameters, this should be available through the initial xml (or if you want to print a simple ascii file, set `NL%IDETAILED=32` in [ED2IN](https://github.com/EDmodel/ED2/blob/master/ED/run/ED2IN) for a complete list of parameters).

If you are modifying the allometric parameters through xml, then make sure the minimum height and minimum DBH are also set in a way that the DBH-Height allometry doesn't crash. This is particularly true for temperate PFTs because the allometric equations (defined in [Albani et al. 2006](https://dx.doi.org/10.1111/j.1365-2486.2006.01254.x)) produce singularities for low height values. At some point it would be worth revisiting these equations, but in the meantime, you can check how the default hgt_min and min_dbh are computed by looking at subroutine ```init_pft_alloc_params``` in [ed_params.f90](https://github.com/EDmodel/ED2/blob/master/ED/src/init/ed_params.f90).

**Reply by Aariq** [reply]
Thanks, that definitely helps.  I was able to calculate the DBH for the `hgt_max` and `hgt_min` using those allometry equations.  I'm curious, what is `hgt_ref`?  It looks like for tropical PFTs the default for `hgt_ref` is about 61.7, but for temperate PFTs it's either 0 or around 1.3.  I can see that being an issue for runs I'm planning to do with temperate grasses using PFT 5.  It seems like it's definitely going to result in singularities.

**Reply by mpaiao** [reply]
The meaning of `hgt_ref` depends on the allometric equation, and there are a few options for tropical PFTs (depending on `NL%IALLOM`). When the Height-DBH allometry uses a Weibull function (e.g., `NL%IALLOM=2`, based on [Poorter _et al._ 2006](https://dx.doi.org/10.1890/0012-9658(2006)87{$[$}1289:AOMTST{$]$}2.0.CO;2)), then `hgt_ref` is the maximum possible height. When the height-DBH allometry uses a log-linear function (e.g., `NL%IALLOM=3`, based on [Longo _et al._ 2020](https://dx.doi.org/10.1029/2020JG005677)), then `hgt_ref` is not used.

**Reply by Aariq** [reply]
Thanks, this was really helpful and looking at the ED2 code helped me figure out the SIGABRT errors I've been getting (https://github.com/EDmodel/ED2/discussions/382) in the process!!  Turns out PEcAn.ED2 was overwriting allometry coefficients to values that correspond to ED2 defaults for `NL%IALLOM=1` while setting `NL%IALLOM=3` in ED2IN (https://github.com/PecanProject/pecan/issues/3124)
